spec/hotel_spec.rb

```diff
@@ -0,0 +1,154 @@
+require_relative 'spec_helper'
+require 'pry'
+
+describe 'Module Hotel' do
+  before do
+    @hotel = Hotel::Hotel.new
+    @start_date = Date.parse("1/9/2018")
+    @end_date = Date.parse("5/9/2018")
+  end
+
+  it 'returns an instance of a room' do
+    room = Hotel::Room.new(1)
+    expect(room).must_be_kind_of Hotel::Room
+  end
+
+  it 'returns array of room number from 1-20' do
+    room_numbers = @hotel.get_all_numbers
+    expect(room_numbers).must_equal [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
+  end
+
+  it 'returns a available room' do
+    room_num = @hotel.get_available_room(@start_date, @end_date)
+    expect(room_num).must_equal 1
+  end
+
+  #Date.parse format: Date.parse("10/5/2018"), 10 is day, 5 is month
+
+  it "raises an argument error for invalid dates" do
+    #proc catches the error when .assigns_a_reservation is invoked
+    proc {
+      @hotel.assigns_a_reservation(@end_date, @start_date)
```

Comment:
This functionality lives in `Reservation` not `Hotel`, so I would test it there.

Reply:
test removed, and it has already been added to Reservation spec.